### PR TITLE
feat(RHINENG-7072) Implement sparse system_profile fields

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-7072](https://issues.redhat.com/browse/RHINENG-7072).
It implements the postgresql version of the `GET /hosts/{host_id_list}/system_profile` endpoint, and adds sparse fields functionality to the other host endpoints (i.e. the `fields` query param works now).

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [x] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by chambridge** [comment]
These are two calls where I'm using with_entities to get a subset of the system profile:
https://github.com/RedHatInsights/insights-host-inventory/blob/master/api/host_query_db.py#L238-L241
https://github.com/RedHatInsights/insights-host-inventory/blob/master/api/host_query_db.py#L265-L267

Not sure if that helps as an example but thought I'd share. Reading through the PR now.

File: app/serialization.py [hunk 1]
```diff
@@ -240,8 +240,15 @@ def serialize_assignment_rule(assign_rule):
     }
 
 
-def serialize_host_system_profile(host):
-    return {"id": _serialize_uuid(host.id), "system_profile": host.system_profile_facts or {}}
+def serialize_host_system_profile(host, sp_fields=[]):
```
Comment (vkrizan) [inline app/serialization.py:243]: `[]` (list) within defaults. I'm surprised that linter did not catch this.

Comment (kruai) [reply app/serialization.py]: good catch, fixed

**Review by chambridge — APPROVED** [review]
LGTM

**Review by FabriciaDinizRH — APPROVED** [review]

**Pull request merged**